[Failing test][sig-cluster-lifecycle] kubeadm-kinder-latest

### Which jobs are failing?

kubeadm-kinder-latest

### Which tests are failing?

* ci-kubernetes-e2e-kubeadm-kinder-latest.Overall
* E2EKubeadm suite.[sig-cluster-lifecycle] [area-kubeadm] proxy addon kube-proxy ServiceAccount should be bound to the system:node-proxier cluster role

### Since when has it been failing?

2021-11-09 23:32 CET

### Testgrid link

https://testgrid.k8s.io/sig-release-master-informing#kubeadm-kinder-latest

### Reason for failure (if possible)

([job run](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kinder-latest/1458473496693706752))
E2EKubeadm suite: [sig-cluster-lifecycle] [area-kubeadm] proxy addon kube-proxy ServiceAccount should be bound to the system:node-proxier cluster role
```bash
_output/local/go/src/k8s.io/kubernetes/test/e2e_kubeadm/proxy_addon_test.go:65
error getting ClusterRoleBindings "kubeadm:node-proxier"
Unexpected error:
    <*errors.StatusError | 0xc0005259a0>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {
                SelfLink: "",
                ResourceVersion: "",
                Continue: "",
                RemainingItemCount: nil,
            },
            Status: "Failure",
            Message: "clusterrolebindings.rbac.authorization.k8s.io \"kubeadm:node-proxier\" not found",
            Reason: "NotFound",
            Details: {
                Name: "kubeadm:node-proxier",
                Group: "rbac.authorization.k8s.io",
                Kind: "clusterrolebindings",
                UID: "",
                Causes: nil,
                RetryAfterSeconds: 0,
            },
            Code: 404,
        },
    }
    clusterrolebindings.rbac.authorization.k8s.io "kubeadm:node-proxier" not found
occurred
_output/local/go/src/k8s.io/kubernetes/test/e2e_kubeadm/util.go:136
```


```bash
   _output/local/go/src/k8s.io/kubernetes/test/e2e_kubeadm/proxy_addon_test.go:65
[AfterEach] [sig-cluster-lifecycle] [area-kubeadm] proxy addon
  /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:186
Nov 10 16:51:29.201: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready

• Failure [0.015 seconds]
[sig-cluster-lifecycle] [area-kubeadm] proxy addon
_output/local/go/src/k8s.io/kubernetes/test/e2e_kubeadm/framework.go:23
  kube-proxy ServiceAccount
  _output/local/go/src/k8s.io/kubernetes/test/e2e_kubeadm/proxy_addon_test.go:60
    should be bound to the system:node-proxier cluster role [It]

[Fail] [sig-cluster-lifecycle] [area-kubeadm] proxy addon kube-proxy ServiceAccount [It] should be bound to the system:node-proxier cluster role 
_output/local/go/src/k8s.io/kubernetes/test/e2e_kubeadm/util.go:136
Ran 31 of 37 Specs in 0.766 seconds
FAIL! -- 30 Passed | 1 Failed | 0 Pending | 6 Skipped
--- FAIL: TestE2E (0.78s)
FAIL

Ginkgo ran 1 suite in 842.396245ms
Test Suite Failed
Error: error running test: exit status 1
 exit status 1

Deleting cluster "kinder-regular" ...
 completed!
Ran 10 of 11 tasks in 0.000 seconds
FAIL! -- 9 tasks Passed | 1 Failed | 1 Skipped

see junit-runner.xml and task logs files for more details

Error: failed executing the workflow 
```

### Anything else we need to know?

This issue seems to be linked to #106311 

FYI: [slack discussion](https://kubernetes.slack.com/archives/CN0K3TE2C/p1636555957038200)

/cc @neolit123 

### Relevant SIG(s)

/sig cluster-lifecycle
/area kubeadm

--- a/cmd/kubeadm/app/phases/addons/proxy/proxy.go
+++ b/cmd/kubeadm/app/phases/addons/proxy/proxy.go
@@ -43,11 +43,11 @@ const (
 	// TODO: This k8s-generic, well-known constant should be fetchable from another source, not be in this package
 	KubeProxyClusterRoleName = "system:node-proxier"
 
+	// KubeProxyClusterRoleBindingName sets the name for the kube-proxy CluterRoleBinding
+	KubeProxyClusterRoleBindingName = "kubeadm:node-proxier"
+
 	// KubeProxyServiceAccountName describes the name of the ServiceAccount for the kube-proxy addon
 	KubeProxyServiceAccountName = "kube-proxy"
-
-	// KubeProxyClusterRoleBindingName sets the name for the kube-proxy CluterRoleBinding
-	KubeProxyClusterRoleBindingName = "kubeam:node-proxier"
 
 	// KubeProxyConfigMapRoleName sets the name of ClusterRole for ConfigMap
 	KubeProxyConfigMapRoleName = "kube-proxy"

--- a/test/e2e_kubeadm/kubelet_config_test.go
+++ b/test/e2e_kubeadm/kubelet_config_test.go
@@ -84,11 +84,15 @@ var _ = Describe("kubelet-config ConfigMap", func() {
 		// https://github.com/kubernetes/kubeadm/issues/1582
 		var UnversionedKubeletConfigMap bool
 		if _, ok := m["featureGates"]; ok {
-			if featureGates, ok := m["featureGates"].(map[string]bool); ok {
+			if featureGates, ok := m["featureGates"].(map[interface{}]interface{}); ok {
 				// TODO: update the default to true once this graduates to Beta.
 				UnversionedKubeletConfigMap = false
 				if val, ok := featureGates["UnversionedKubeletConfigMap"]; ok {
-					UnversionedKubeletConfigMap = val
+					if valBool, ok := val.(bool); ok {
+						UnversionedKubeletConfigMap = valBool
+					} else {
+						framework.Failf("unable to cast the value of feature gate UnversionedKubeletConfigMap to bool")
+					}
 				}
 			} else {
 				framework.Failf("unable to cast the featureGates field in the %s ConfigMap", kubeadmConfigName)

